src/libraries/Microsoft.Extensions.Logging.Console/src/WindowsLogConsole.cs

```diff
@@ -61,5 +127,77 @@ public void Flush()
             // No action required as for every write, data is sent directly to the console
             // output stream
         }
+
+        private static bool SetsForegroundColor(int number, bool isDark, out ConsoleColor? color)
```

Comment:
>SetsForegroundColor [](start = 28, length = 19)

this and SetsBackgroundColor can be simple array lookup instead of switch statement.

Reply:
/cc: @adamsitnik

Reply:
it would be a tradeoff - we would increase the memory footprint but reduced the code size

afair there was a feature request to make this colors configurable. In that case, we might just use dictionary in the future